engines/private/private.cpp

```diff
@@ -0,0 +1,1120 @@
+
+    // Initialize stuff
+    initInsts();
+    initFuncs();
+    initCursors();
+
+    parse((char *) buf);
+    free(buf);
+    delete file;
+    assert(constants.size() > 0);
+
+    // Initialize graphics using following:
+    _screenW = 640;
+    _screenH = 480;
+    //_pixelFormat = Graphics::PixelFormat::createFormatCLUT8();
+    _pixelFormat = Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0);
+    _transparentColor = _pixelFormat.RGBToColor(0,255,0);
+    initGraphics(_screenW, _screenH, &_pixelFormat);
+    changeCursor("default");
+
+    _origin = new Common::Point(0, 0);
+    _image = new Image::BitmapDecoder();
+    _compositeSurface = new Graphics::ManagedSurface();
+    _compositeSurface->create(_screenW, _screenH, _pixelFormat);
+    _compositeSurface->setTransparentColor(_transparentColor);
+
+    // Create debugger console. It requires GFX to be initialized
+    Console *console = new Console(this);
+    setDebugger(console);
+
+    // Additional setup.
+    debug("PrivateEngine::init");
+
+    // Simple main event loop
+    Common::Event event;
+    Common::Point mousePos;
+    _videoDecoder = nullptr; //new Video::SmackerDecoder();
+
+    int saveSlot = ConfMan.getInt("save_slot");
+    if (saveSlot >= 0) { // load the savegame
+        loadGameState(saveSlot);
+    } else {
+        _nextSetting = &kGoIntro;
+    }
+
+    while (!shouldQuit()) {
+        checkPhoneCall();
+
+        while (g_system->getEventManager()->pollEvent(event)) {
+            mousePos = g_system->getEventManager()->getMousePos();
+            // Events
+            switch (event.type) {
+            case Common::EVENT_KEYDOWN:
+                if (event.kbd.keycode == Common::KEYCODE_ESCAPE && _videoDecoder)
+                    skipVideo();
+
+                break;
+
+            case Common::EVENT_QUIT:
+            case Common::EVENT_RETURN_TO_LAUNCHER:
+                break;
+
+            case Common::EVENT_LBUTTONDOWN:
+                _numberClicks++;
+                if (selectDossierNextSuspect(mousePos))
+                    break;
+                else if (selectDossierPrevSuspect(mousePos))
+                    break;
+
+                selectPauseMovie(mousePos);
+                selectPhoneArea(mousePos);
+                selectPoliceRadioArea(mousePos);
+                selectAMRadioArea(mousePos);
+                selectLoadGame(mousePos);
+                selectSaveGame(mousePos);
+                if (!_nextSetting)
+                    selectMask(mousePos);
+                if (!_nextSetting)
+                    selectExit(mousePos);
+                break;
+
+            case Common::EVENT_MOUSEMOVE:
+                changeCursor("default");
+                if      (cursorPauseMovie(mousePos)) {}
+                else if (cursorMask(mousePos))       {}
+                else if (cursorExit(mousePos))       {}
+                //
+                break;
+
+            default:
+            {}
```

Comment:
huh?

Reply:
Avoid that default causes clang to give a very long warning:

```
engines/private/private.cpp: In member function 'virtual Common::Error Private::PrivateEngine::run()':
engines/private/private.cpp:201:20: warning: enumeration value 'EVENT_INVALID' not handled in switch [-Wswitch]
  201 |             switch (event.type) {
      |                    ^
engines/private/private.cpp:201:20: warning: enumeration value 'EVENT_KEYUP' not handled in switch [-Wswitch]
engines/private/private.cpp:201:20: warning: enumeration value 'EVENT_LBUTTONUP' not handled in switch [-Wswitch]
engines/private/private.cpp:201:20: warning: enumeration value 'EVENT_RBUTTONDOWN' not handled in switch [-Wswitch]
engines/private/private.cpp:201:20: warning: enumeration value 'EVENT_RBUTTONUP' not handled in switch [-Wswitch]
....
```

I didn't want to polute my warnings with that. Is there an alternative way to do it without defining each case?

Reply:
I am referring to `{}`. The canonical way is to put `break;` there.

Reply:
Oh, I was just extending this code convetion rule for switch:

> Mandatory {} for empty for/while loops

Anyway, I change it to use `break`.